README
mobilAP Version 2.0b14
May 23, 2010

RELEASE NOTES

This is (hopefully) the final Beta version of mobilAP 2.0! Please see the CHANGELOG file for changes.

It is now considered appropriate for production use. Most of the important features are functional, however, there may still be some unresolved issues. We would appreciate you using this in a non-critical setting and providing feedback to mobilAP@daap.uc.edu. 

REQUIREMENTS AND SETUP

- Please read the Documentation for system requirements and setup information. Open up index.html in the Documentation folder of this archive.

KNOWN ISSUES/PLANNED UPDATES BEFORE 2.0 final

- Due to a coding flaw in Apple's Dashcode framework code, if you edit the project file and then re-publish the site, the desktop version will not work in Firefox (and perhaps other browsers). The template used to program the Stack Layout part calls a method that does not exist in Firefox. In the code that comes with mobilAP, that method is properly handled. If you republish the project it will overwrite that file (html/safari/Parts/parts.js).

Please contact mobilap@daap.uc.edu if you have any questions, bug submissions or requests. We make no guarantee, however, with respect to future releases.

Please note that technical support cannot be given, although paid consultation for help is available.

Monitor http://daap.uc.edu/mobilap or @mobilAP on twitter for future updates.

The license is a very liberal BSD Style license. You may alter this as you need as long as you continue to give original attribution to the University of Cincinnati. See the LICENSE file for more information


CHANGELOG
2.0b14 (May 23, 2010)
- Fixed an issue where the wrong session presenters would show up on the mobile site

2.0b13 (May 13, 2010)
- Added the ability to view session evaluation results
- Added the ability to clear evaluations for a given session

2.0b12 (April 26, 2010)
- Fixed regression when attempting to manually add/edit user
- Make sure that only admins can clear question results
- Fixed regression where placeholder image would not show in mobile site
- Fixed issue where presenters don't display on mobile site

2.0b11 (April 19, 2010)
- Documentation has been included, load the index.html file in the Documentation folder. Please provide feedback on the documentation
- Renamed "root" folder to mobilAP
- Moved import files to import folder
- Announcement posted information will now show properly
- Other minor fixes discovered while writing documentation

2.0b10 (April 8, 2010)
- You can now search the directory. Searches filter by name or organization
- The mobile "header" now reflects the site title as specified
- After uploading a picture for a user, the image updates properly
- You can now upload a webclip icon using the admin interface.
- You can now import users using a tab delimited file. See the included example_import files for field order
- Bar charts are now available. Simply click/tap the chart to switch between bar and pie charts. Hopefully some
intrepid designer can help make these look more snazzy.


2.0b9 (March 25, 2010)
- If you don't want to allow write access to the data folder, you can easily edit the database configuration yourself - NOTE: without write access you will be unable to save uploaded directory photos
- You can instruct mobilAP to use a different folder for the SQLite database
- Situations where you have partially completed setup and return to setup are handled better.
- You can (and indeed, must) set the event timezone during setup
- You can change the labels and visibility of the items on the home screen/list (see Home List tab in Admin)

2.0b8 (March 21, 2010)
- Added some indexes to improve database performance (not critical for low volume sites, but still useful)
- Updated serial numbering system to better handle user submitted content (viewed announcements, question answering, etc)
- Fixed an issue where setup may give errors after selecting MySQL during setup
- Fixed issues with unauthorized errors when saving data during setup
- removed some dissolve transitions that could cause the browser to hide content
- Improved creating and editing evaluation questions
- Removed some old console logging that is now unnecessary
- Allow deletion of sessions (which also deletes any schedule items that use the session)
- Resize the directory window in the desktop version to fit the entire height of the browser window
- Updated CSS to provide more font alternatives for non-safari browsers
- Added the ability to make session questions inactive (i.e. users do not see them, but admins do)

2.0b7 (March 11, 2010)
- Fixed issues that allowed clever people to perform administrative functions without authorization
- Fixed an issue with editing evaluation questions
- Properly handled sites with private content

2.0b6 (March 2, 2010)
- ADDITION: Implemented the serial numbering system that dramatically reduces network reloads by detecting whether a given set of data has been updated. This is crucial on the iPhone version since parsing and then updating the UI can be very expensive depending on the data set. This allows more frequent updates (5 sec) but will only pull down and parse data if it's new. 

2.0b5 (March 1, 2010)
- Fixed an issue that would appear when mobilAP was placed on a case sensitive file system
- Some JSON values should be returned as numbers rather than strings
- Added announcement read/unread methods (not fully implemented)
- Added framework to enable updating of data only when it has been changed saving bandwidth and processing time. This has not been fully implemented yet
- Fixed an issue when an error was given if a user was updated but no picture was uploaded (admin)
- updated discussion formatting (desktop)
- Discussion is now sorted ascending (i.e. most recent at the bottom)
- Fixed formatting issue when editing announcements
- Optimized some unnecessary data loads
- removed dissolve when changing session "tabs" behaves more like native app
- fixed formatting issue with long titles/detail in schedule (mobile)
- changed session reload timer to 15 seconds (will eventually altered to only reload when data has changed)

2.0b4 (February 2, 2010)
- Fixed an issue that would prevent the mobilAP setup from working when short_open_tags was not enabled
- Added character encoding to output.

2.0b3 (November 25, 2009)
- Fixed an issue that would prevent mobilAP from working in PHP versions less than 5.3

2.0b2 (November 18, 2009)
- Evaluation questions can be added by admins and answered by users
- Config files and welcome content have been moved to html/mobilAP/data. This was done so you only have to open up 1 folder that must be writable.
- Setup files were rearranged.
- Directory and Announcements are refreshed every 60 seconds
- Administrators can upload directory photos
- EXIF orientation is removed to account for the difference in how browsers handle it. In order for this to work requires PHP's EXIF and GD functions. If those functions aren't available, you will need to rotate the images manually.
- If passwords are optional the password box will not show
- The content private setting now works as expected

2.0b1 (November 4, 2009)
- Initial Release


html/Parts/setup.js
/* 
 This file was generated by Dashcode and is covered by the 
 license.txt included in the project.  You may edit this file, 
 however it is recommended to first turn off the Dashcode 
 code generator otherwise the changes will be lost.
 */
var dashcodePartSpecs = {
    "announcement_list_title": { "creationFunction": "CreateText", "text": "Item" },
    "announcement_text": { "creationFunction": "CreateText", "text": "Announcement Text" },
    "announcement_timestamp": { "creationFunction": "CreateText", "text": "Posted: x" },
    "announcement_title": { "creationFunction": "CreateText", "text": "Announcement Title" },
    "announcements_list": { "creationFunction": "CreateList", "dataArray": ["Item 1", "Item 2", "Item 3"], "dataSourceName": "announcement_controller", "labelElementId": "announcement_list_title", "listStyle": "List.EDGE_TO_EDGE", "sampleRows": 3, "useDataSource": true },
    "announcements_title": { "creationFunction": "CreateText", "text": "Announcements" },
    "back_button": { "creationFunction": "CreatePushButton", "initialHeight": 30, "initialWidth": 60, "leftImageWidth": 16, "rightImageWidth": 5, "text": "Back" },
    "back_to_questions": { "creationFunction": "CreatePushButton", "initialHeight": 30, "initialWidth": 302, "leftImageWidth": 15, "onclick": "session_question_answers.back_to_questions", "rightImageWidth": 15, "text": "Back to Questions" },
    "browser": { "creationFunction": "CreateBrowser" },
    "demo_text": { "creationFunction": "CreateText" },
    "detail_name": { "creationFunction": "CreateText", "text": "Detail Name" },
    "detail_text": { "creationFunction": "CreateText", "text": "Text" },
    "directory_bio": { "creationFunction": "CreateText" },
    "directory_detail_dept": { "creationFunction": "CreateText", "text": "dept" },
    "directory_detail_email": { "creationFunction": "CreateText", "text": "email" },
    "directory_detail_name": { "creationFunction": "CreateText", "text": "Name" },
    "directory_detail_organization": { "creationFunction": "CreateText", "text": "organization" },
    "directory_detail_title": { "creationFunction": "CreateText", "text": "title" },
    "directory_letters_list": { "creationFunction": "CreateList", "dataArray": ["Item 1", "Item 2", "Item 3"], "dataSourceName": "directoryLetters", "labelElementId": "letter", "listStyle": "List.EDGE_TO_EDGE", "sampleRows": 3, "useDataSource": true },
    "directoryList": { "creationFunction": "CreateList", "dataArray": ["Item 1", "Item 2", "Item 3"], "dataSourceName": "directoryController", "labelElementId": "name", "listStyle": "List.EDGE_TO_EDGE", "sampleRows": 5, "useDataSource": true },
    "evaluation_intro": { "creationFunction": "CreateText", "text": "We appreciate your feedback. Please answer the following brief questions and include additional comments if you wish." },
    "evaluation_thanks_text": { "creationFunction": "CreateText", "text": "Thank you for your response. We appreciate your feedback." },
    "generic_list_list": { "creationFunction": "CreateList", "dataArray": ["Item 1", "Item 2", "Item 3"], "dataSourceName": "genericListController", "labelElementId": "genericList_label", "listStyle": "List.EDGE_TO_EDGE", "sampleRows": 3, "useDataSource": true },
    "genericList_label": { "creationFunction": "CreateText", "text": "Item" },
    "header": { "creationFunction": "CreateHeader", "rootTitle": "mobilAP" },
    "homeList": { "creationFunction": "CreateList", "dataSourceName": "browserController", "labelElementId": "listTitle", "listStyle": "List.EDGE_TO_EDGE", "sampleRows": 4, "useDataSource": true },
    "letter": { "creationFunction": "CreateText", "text": "Item" },
    "link_title_label": { "creationFunction": "CreateText", "text": "Title:" },
    "link_url_label": { "creationFunction": "CreateText", "text": "URL:" },
    "links_save_button": { "creationFunction": "CreatePushButton", "initialHeight": 30, "initialWidth": 302, "leftImageWidth": 15, "onclick": "session_links.submit", "rightImageWidth": 15, "text": "Add Link" },
    "listTitle": { "creationFunction": "CreateText", "text": "Item" },
    "login_button": { "creationFunction": "CreatePushButton", "initialHeight": 30, "initialWidth": 60, "leftImageWidth": 5, "onclick": "mobilAP.login_button_handler", "rightImageWidth": 5, "text": "Login" },
    "login_result": { "creationFunction": "CreateText" },
    "login_submit": { "creationFunction": "CreatePushButton", "initialHeight": 25, "initialWidth": 60, "leftImageWidth": 5, "onclick": "mobilAP.login_submit", "rightImageWidth": 5, "text": "Login" },
    "name": { "creationFunction": "CreateText", "text": "Name" },
    "organization": { "creationFunction": "CreateText", "text": "organization" },
    "post_submit": { "creationFunction": "CreatePushButton", "initialHeight": 30, "initialWidth": 128, "leftImageWidth": 15, "onclick": "session_chat.submit", "rightImageWidth": 15, "text": "Submit Post" },
    "post_text": { "creationFunction": "CreateText", "text": "Item" },
    "post_timestamp": { "creationFunction": "CreateText", "text": "time" },
    "post_user": { "creationFunction": "CreateText", "text": "First Last" },
    "post_view": { "creationFunction": "CreatePushButton", "initialHeight": 30, "initialWidth": 131, "leftImageWidth": 15, "onclick": "session_chat.view_posts", "rightImageWidth": 15, "text": "View Posts" },
    "presenter_name": { "creationFunction": "CreateText", "text": "Name" },
    "presenter_organization": { "creationFunction": "CreateText", "text": "Organization" },
    "programList_detail": { "creationFunction": "CreateText", "text": "Event detail" },
    "programList_room": { "creationFunction": "CreateText", "text": "Room" },
    "programList_time": { "creationFunction": "CreateText", "text": "07:00" },
    "programList_title": { "creationFunction": "CreateText", "text": "Event Title" },
    "programs_list": { "creationFunction": "CreateList", "dataArray": ["Item 1", "Item 2", "Item 3"], "dataSourceName": "programSchedule", "labelElementId": "programList_time", "listStyle": "List.ROUNDED_RECTANGLE", "sampleRows": 3, "useDataSource": true },
    "question_answers": { "creationFunction": "CreateList", "dataArray": ["Item 1", "Item 2", "Item 3"], "dataSourceName": "session_question_answers", "labelElementId": "question_response_text_label", "listStyle": "List.ROUNDED_RECTANGLE", "sampleRows": 3, "useDataSource": true },
    "question_response_count": { "creationFunction": "CreateText", "text": "20" },
    "question_response_index": { "creationFunction": "CreateText", "text": "1." },
    "question_response_label": { "creationFunction": "CreateText", "text": "Response" },
    "question_response_text": { "creationFunction": "CreateText", "text": "Question Text" },
    "question_response_text_label": { "creationFunction": "CreateText", "text": "Item" },
    "question_response_total": { "creationFunction": "CreateText", "text": "X Responses" },
    "question_responses": { "creationFunction": "CreateList", "dataArray": ["Item 1", "Item 2", "Item 3"], "dataSourceName": "session_question", "labelElementId": "question_response_label", "listStyle": "List.ROUNDED_RECTANGLE", "sampleRows": 3, "useDataSource": true },
    "question_text": { "creationFunction": "CreateText", "text": "Question Text" },
    "session_abstract": { "creationFunction": "CreateText", "text": "Loading..." },
    "session_add_link_button": { "creationFunction": "CreatePushButton", "initialHeight": 30, "initialWidth": 302, "leftImageWidth": 15, "onclick": "session.show_add_link", "rightImageWidth": 15, "text": "Add Link" },
    "session_data_stack": { "creationFunction": "CreateStackLayout", "subviewsTransitions": [{ "direction": "right-left", "duration": ".55", "timing": "ease-in-out", "type": "cube" }, { "direction": "right-left", "duration": ".55", "timing": "ease-in-out", "type": "cube" }, { "direction": "bottom-top", "duration": ".55", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": ".55", "timing": "ease-in-out", "type": "cube" }, { "direction": "right-left", "duration": ".55", "timing": "ease-in-out", "type": "cube" }, { "direction": "right-left", "duration": ".55", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": ".55", "timing": "ease-in-out", "type": "cube" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }] },
    "session_day": { "creationFunction": "CreateText", "text": "Loading Schedule..." },
    "session_discussion_count": { "creationFunction": "CreateText" },
    "session_discussion_description": { "creationFunction": "CreateText", "text": "Talk about this session here" },
    "session_discussion_list": { "creationFunction": "CreateList", "dataArray": ["Item 1", "Item 2", "Item 3"], "dataSourceName": "session_chat", "labelElementId": "post_timestamp", "listStyle": "List.ROUNDED_RECTANGLE", "sampleRows": 3, "useDataSource": true },
    "session_discussion_next": { "creationFunction": "CreatePushButton", "initialHeight": 30, "initialWidth": 128, "leftImageWidth": 15, "onclick": "session_chat.nextPage", "rightImageWidth": 15, "text": "Previous Posts" },
    "session_discussion_post": { "creationFunction": "CreatePushButton", "initialHeight": 30, "initialWidth": 301, "leftImageWidth": 15, "rightImageWidth": 15, "text": "New Post" },
    "session_discussion_prev": { "creationFunction": "CreatePushButton", "initialHeight": 30, "initialWidth": 127, "leftImageWidth": 15, "onclick": "session_chat.prevPage", "rightImageWidth": 15, "text": "Recent Posts" },
    "session_group_list": { "creationFunction": "CreateList", "dataArray": ["Item 1", "Item 2", "Item 3"], "dataSourceName": "session_group", "labelElementId": "sessionGroupList_time", "listStyle": "List.ROUNDED_RECTANGLE", "sampleRows": 3, "useDataSource": true },
    "session_group_title": { "creationFunction": "CreateText", "text": "Session Group" },
    "session_links_description": { "creationFunction": "CreateText", "text": "Here you can find and post links that are relevant to this session." },
    "session_links_label": { "creationFunction": "CreateText", "text": "Item" },
    "session_links_list": { "creationFunction": "CreateList", "dataArray": ["Link 1", "Link 2", "Link 3"], "dataSourceName": "session_links", "labelElementId": "session_links_label", "listStyle": "List.ROUNDED_RECTANGLE", "sampleRows": 3, "useDataSource": true },
    "session_menu_label": { "creationFunction": "CreateText", "text": "Item" },
    "session_menu_list": { "creationFunction": "CreateList", "dataArray": ["Wednesday", "Thursday", "Friday"], "dataSourceName": "session_days", "labelElementId": "session_menu_label", "listStyle": "List.ROUNDED_RECTANGLE", "sampleRows": 3, "useDataSource": true },
    "session_presenters_list": { "creationFunction": "CreateList", "dataArray": ["Item 1", "Item 2", "Item 3"], "dataSourceName": "session_presenters", "labelElementId": "presenter_name", "listStyle": "List.ROUNDED_RECTANGLE", "sampleRows": 3, "useDataSource": true },
    "session_question_num": { "creationFunction": "CreateText", "text": "1." },
    "session_question_text": { "creationFunction": "CreateText", "text": "Question" },
    "session_questions_list": { "creationFunction": "CreateList", "dataArray": ["Question 1", "Question 2", ["Question 3", "Question 3"]], "dataSourceName": "session_questions", "labelElementId": "session_question_num", "listStyle": "List.ROUNDED_RECTANGLE", "sampleRows": 3, "useDataSource": true },
    "session_rate_button": { "creationFunction": "CreatePushButton", "initialHeight": 30, "initialWidth": 302, "leftImageWidth": 15, "onclick": "session.start_evaluation", "rightImageWidth": 15, "text": "Rate this session" },
    "session_title": { "creationFunction": "CreateText", "text": "Session Page" },
    "sessionGroupList_detail": { "creationFunction": "CreateText", "text": "Text" },
    "sessionGroupList_room": { "creationFunction": "CreateText", "text": "room" },
    "sessionGroupList_title": { "creationFunction": "CreateText", "text": "Event Title" },
    "sessions_current_detail": { "creationFunction": "CreateText", "text": "Text" },
    "sessions_current_heading": { "creationFunction": "CreateText", "text": "Current Sessions" },
    "sessions_current_list": { "creationFunction": "CreateList", "dataArray": ["Item 1", "Item 2", "Item 3"], "dataSourceName": "current_sessions", "labelElementId": "sessions_current_time", "listStyle": "List.ROUNDED_RECTANGLE", "sampleRows": 2, "useDataSource": true },
    "sessions_current_room": { "creationFunction": "CreateText", "text": "Text" },
    "sessions_current_time": { "creationFunction": "CreateText", "text": "Item" },
    "sessions_current_title": { "creationFunction": "CreateText", "text": "Text" },
    "stackLayout": { "creationFunction": "CreateStackLayout", "subviewsTransitions": [{ "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }, { "direction": "right-left", "duration": "", "timing": "ease-in-out", "type": "push" }] },
    "submit_evaluation_next": { "creationFunction": "CreatePushButton", "customImagePosition": "PushButton.IMAGE_POSITION_NONE", "initialHeight": 30, "initialWidth": 45, "leftImageWidth": 15, "onclick": "session_evaluation.next", "rightImageWidth": 15, "text": "Next" },
    "submit_evaluation_prev": { "creationFunction": "CreatePushButton", "initialHeight": 30, "initialWidth": 45, "leftImageWidth": 15, "onclick": "session_evaluation.previous", "rightImageWidth": 15, "text": "Previous" },
    "submit_response": { "creationFunction": "CreatePushButton", "initialHeight": 30, "initialWidth": 45, "leftImageWidth": 15, "onclick": "session_question.submit", "rightImageWidth": 15, "text": "Submit Response" },
    "view_results": { "creationFunction": "CreatePushButton", "initialHeight": 30, "initialWidth": 45, "leftImageWidth": 15, "onclick": "session_question.view_results", "rightImageWidth": 15, "text": "View Results" },
    "welcomeHeading": { "creationFunction": "CreateText", "text": "Welcome" },
    "welcomeText": { "creationFunction": "CreateText", "text": "" }
};
